migrations/lib/converters/discourse/steps/categories.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+module Migrations::Converters::Discourse
+  class Categories < ::Migrations::Converters::Base::ProgressStep
+    attr_accessor :source_db
+
+    def max_progress
+      @source_db.count <<~SQL
+        SELECT COUNT(*) FROM categories
+      SQL
+    end
+
+    def items
+      skip_existing_ids = settings.dig(:categories, :skip_existing_ids) ? "TRUE" : "FALSE"
+
+      @source_db.query <<~SQL
+        WITH seeded_categories AS (
+                                    SELECT value::int as category_id, name as setting_name
+                                    FROM site_settings
+                                    WHERE name IN (
+                                      'uncategorized_category_id',
+                                      'meta_category_id',
+                                      'staff_category_id',
+                                      'general_category_id'
+                                    )
+        )
+        SELECT c.*,
+               CASE WHEN #{skip_existing_ids} THEN NULL ELSE sc.setting_name END AS existing_id
+        FROM categories c
+             LEFT JOIN seeded_categories sc ON c.id = sc.category_id;
+      SQL
```

Comment:
The PG adapter supports parameter binding.

```suggestion
      @source_db.query(<<~SQL, skip_existing_ids)
        WITH seeded_categories AS (
                                    SELECT value::int as category_id, name as setting_name
                                    FROM site_settings
                                    WHERE name IN (
                                      'uncategorized_category_id',
                                      'meta_category_id',
                                      'staff_category_id',
                                      'general_category_id'
                                    )
        )
        SELECT c.*,
               CASE WHEN $1 THEN NULL ELSE sc.setting_name END AS existing_id
        FROM categories c
             LEFT JOIN seeded_categories sc ON c.id = sc.category_id;
      SQL
```